themes/haystack/layouts/_default/tutorials.html

```diff
@@ -194,10 +198,179 @@ <h2>{{ .Params.contribute.title }}</h2>
       {{/*  document.querySelector("#task-select").value = "all";  */}}
       filterAndSort();
     }
+    
+    {{/*  Modal  */}}
+    const searchModalContainer = document.querySelector(".search-modal-container");
+    const searchModal = document.querySelector(".search-modal");
+    const searchModalInput = document.querySelector("#modal-search");
+    const searchModalCards = document.querySelector(".search-modal-cards");
+    
+    {{/*  Open modal  */}}
+    const openModal = () => {
+      searchModalContainer.classList.add("is-active");
+
+      const body = document.querySelector("body");
+      const scrollbarWidth = window.innerWidth - body.clientWidth;
+      body.style.overflow = "hidden";
+
+      if (window.innerWidth > 576){
+        body.style.paddingRight = scrollbarWidth + "px";
+        searchModalContainer.style.paddingRight = scrollbarWidth + "px";
+      }
+
+      searchModalInput.focus();
+    };
+
+    {{/*  Close modal  */}}
+    const closeModal = () => {
+      searchModalContainer.classList.remove("is-active");
+      
+      const body = document.querySelector("body");
+      body.style.overflow = "auto";
+      body.style.paddingRight = 0;
+      searchModalContainer.style.paddingRight = 0;
+
+      setTimeout(() => {
+        searchModalInput.value = ""
+        searchModal.classList.remove("not-found", "loading")
+        searchModal.classList.add("default")
+        searchModalCards.innerHTML = ""
+      }, 200)
+
+    };
+
+    {{/*  Close modal on esc  */}}
+    document.addEventListener("keydown", (e) => {
+      if (e.keyCode === 27) {
+        closeModal();
+      }
+    });
+
+    {{/*  Close modal on outside click  */}}
+    searchModalContainer.addEventListener("click", (e) => {
+      if (e.target === searchModalContainer) {
+        closeModal();
+      }
+    });
+
+    {{/*  Search  */}}
+    const searchTutorials = async (query) => {
+      searchModal.classList.remove("default", "not-found")
+      searchModalCards.innerHTML = ""
+      searchModal.classList.add("loading")
+
+      try {
+        const response = await fetch(`/api/tutorial-search?query=${query}`);
+
+        if (response.status === 200) {
+          const result = await response.json();
+
+          if (result.results[0].documents.length > 0) {
+            {{/*  build cards  */}}
+            result.results[0].documents.forEach((doc) => {
+              const card = document.createElement("a")
+              card.classList.add("search-modal-card")
+              card.href = `/tutorials/${doc.meta.file_name.toLowerCase().replace(".txt", "")}`;
```

Comment:
I updated the tutorial response. Now, every tutorial has `headlines` information in the `meta` field. Can we use this information to direct the users precisely to the place where the text is retrieved from? 

**Here's an example**
This is a headline of a result:
```
headlines: [
0: {level: 1, headline: "Initializing the PromptNode"}
1: {level: 1, start_idx: 455, headline: "Defining the Pipeline"}
]
```
This is the `meta.link` of the same result: `tutorials/22_pipeline_with_promptnode`
The link that this result item directs to should be: `tutorials/22_pipeline_with_promptnode#initializing-the-promptnode` 
Basically, you need to take only the first item of headlines and add it to the end of the link with `-` instead of space.